APCAppCore/APCAppCore/Library/APCTaskImporter.m

```diff
@@ -191,6 +194,56 @@ - (BOOL) processTasks: (NSArray *) arrayOfTasks
         }
     }
     
+    // -----------------------------------------------------
+    // Delete Obsolete Tasks
+    // -----------------------------------------------------
+    // Relevant tasks should have come down from bridge and then update the APCTask.updatedAt field
+    // accordingly with the exception of: completed tasks and expired tasks. Any task that wasn't updated
+    // that doesn't fall into one of those 2 categories is either no longer scheduled or at least it is
+    // no longer relevant. So lets delete those. If at some point Bridge starts returning recently completed
+    // and recently expired activities, this will either be unnecessary or be significantly simpler.
+    if (success) {
+        // Tasks saved, so now clean out the tasks that didn't come down from the server
+        NSDate *now = [NSDate date];
+        NSTimeInterval fiveMinutesAgoInterval = 60 * 5 * (-1);
+        NSDate *fiveMinutesAgo = [now dateByAddingTimeInterval:fiveMinutesAgoInterval];
+        NSDate *midnightThisMorning = now.startOfDay;
+        NSDate *midnightYesterdayMorning = [now dateByAddingDays:(-1)].startOfDay;
+        
+        
+        // TODO: This logic can be simplified once daysBehind is an option - we no longer have to worry about deleting
+        // "Yesterday" tasks.
+        // Check that the activity was not finished - if it was, we don't want to remove it (since Bridge stops returning it)
+        // Check that the activity was not a "yesterday" task, by confirming it:
+            // Had no expiration OR
+            // Expired before yesterday OR
+            // Expired today or later
+        NSPredicate *findObsoleteTasks = [NSPredicate predicateWithFormat:
+                                          @"(%K < %@) && (%K == nil) && (%K.length == 0 OR %K < %@ OR %K > %@)",
+                                          NSStringFromSelector (@selector (updatedAt)),           // -[APCTask taskScheduledFor]
+                                          fiveMinutesAgo,
+                                          NSStringFromSelector (@selector (taskFinished)),           // -[APCTask taskExpires]
+                                          NSStringFromSelector (@selector (taskExpires)),           // -[APCTask taskExpires]
+                                          NSStringFromSelector (@selector (taskExpires)),           // -[APCTask taskExpires]
+                                          midnightYesterdayMorning,
```

Comment:
FPHS doesn't care about previously expired tasks as it doesn't have a 'yesterday' section equivalent. The main difference w/ displaying tasks in FPHS is it shows finished tasks back to the start of the week instead of only stuff finished today.